src/DebuggingFeature.ts

```diff
@@ -0,0 +1,80 @@
+import { BaseLanguageClient, StaticFeature, Neovim } from "coc.nvim";
+import {
+  ExecuteCommandParams,
+  ExecuteCommandRequest,
+  InitializeParams,
+  ServerCapabilities
+} from "vscode-languageserver-protocol";
+import { commands, workspace } from "coc.nvim";
+
+interface DebugAdapterStartResponse {
+  name: string;
+  uri: string;
+}
+
+export class DebuggingFeature implements StaticFeature {
+  private withVimspector = false;
+
+  constructor(private _nvim: Neovim, private _client: BaseLanguageClient) {}
+
+  async preInit(): Promise<void> {
+    this.withVimspector = (await this._nvim.getVar("loaded_vimpector")) === 1;
+  }
+
+  fillInitializeParams(params: InitializeParams): void {
+    if (workspace.isNvim && this.withVimspector) {
```

Comment:
Can we explicitly set `debuggingProvider` to false is this is false?

Reply:
Done.